hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestHDFSConcat.java

```diff
@@ -564,4 +564,36 @@ public void testConcatOnSameFile() throws Exception {
 
     assertEquals(1, dfs.getContentSummary(new Path(dir)).getFileCount());
   }
+
+  /**
+   * Verifies concat with wrong user when dfs.permissions.enabled is false.
+   *
+   * @throws IOException
+   */
+  @Test
+  public void testConcatPermissions() throws IOException {
```

Comment:
Please refer: org.apache.hadoop.security.TestPermission to build the unit tests. IMO, we should add at least one case which can pass and another one case that failed since permission can not pass. JFYI, Thanks.

Reply:
@Hexiaoqiao In TestHDFSConcat::testConcat(), there are a failed permission case. And I also add a failed case in testConcatPermissions(). Please help to review.

Reply:
1. We should not setup cluster and destroy it again and again. Actually for `TestHDFSConcat`,  it is ready to setup cluster and also it can destroy it automatically. Please refer to annotation `@Before` and `@After`.
2. Suggest to set permission explicitly before invoke concat then it will be more readable and clearly.
3. Please check Yetus report carefully include failed unit test and checkstyles.
Thanks again.

Reply:
Without MiniDFSCluster restart, can we change the value of dfs.permissions.enabled dynamically?

Reply:
@Hexiaoqiao I have tried to set permission explicitly before invoke concat, but the permission config failed take effect. Could you tell me how to change dfs.permissions.enabled dynamically?